### PR TITLE
fix(website): fix issue with change website type failed

### PR DESCRIPTION
**Pull request by zhengkunwang223**

### CODE REVIEW

File: agent/app/service/website.go [hunk 1]
```diff
@@ -1383,11 +1386,18 @@ func (w WebsiteService) ChangePHPVersion(req request.WebsitePHPVersionReq) error
...
 		website.RuntimeID = 0
 		website.Type = constant.Static
 		website.Proxy = ""
 		server.RemoveDirective("location", []string{"~", "[^/]\\.php(/|$)"})
+		if !fileOp.Stat(indexHtmlPath) {
+			_ = fileOp.WriteFile(indexHtmlPath, strings.NewReader(string(nginx_conf.Index)), constant.FilePerm)
+		}
 	}
 
 	config.FilePath = configPath
```
Comment (wanghe-fit2cloud) [inline agent/app/service/website.go:1403]: Yes, there are some improvements to be made here. Here's an revised version:

1. **Function Definition**:
   Ensure that the function parameters match those expected in `request.WebsitePHPVersionReq`.
   
2. **File Path Calculation**:
   The site paths should be consistent within the same context (e.g., use `GetSitePath` consistently).

3. **Error Checking**:
   Add more detailed error messages for better debugging.

4. **Simplified Logic**:
   Combine repeated logic into helper functions when necessary to improve readability and maintainability.

5. **Permissions Management**:
   Consider adding checks or ensuring proper permissions if writing changes to static files is required.

Below is the updated code snippet with these enhancements:

```go
func (w WebsiteService) ChangePHPVersion(req request.WebsitePHPVersionReq) error {
    server := w.getServerByWebsiteID(req.WebsiteID)

    if _, err := os.Stat(configFilePath); err != nil && !os.IsNotExist(err) {
        log.Fatalf("Failed to open NGINX configuration file at %s: %v", configFilePath, err)
        return err
    }

    // Load existing nginx.conf
    bytes, err := ioutil.ReadFile(configFilePath)
    if err != nil {
        log.Fatalf("Failed to read NGINX configuration file from local directory: %v", err)
        return err
    }

    servers := nginxConf.LoadServers(bytes)

    if len(servers) == 0 {
        return errors.New("nginx config is not valid")
    }
    
    server := servers[0]

    if req.RuntimeID > 0 {
        server.UpdateDirective("index", []string{"index.php", "index.html", "index.htm", "default.php", "default.htm", "default.html"})
        server.RemoveDirective("location", []string{"~", "[^/]\\.php(/|$)"})

        runtime, err := runtimeRepo.GetFirst(repo.WithByID(req.RuntimeID))
        if err != nil {
            return err
        }
        
        website.Proxy = webRuntimeToNginxProxy(runtime)
        server.UpdatePHPProxy(webRuntimeToNginxProxy(runtime), "")

        path := GetStaticSitePath(req.WebsiteID, SiteIndexDir)
        fileOp := files.NewFileOp()

        indexFiles := []string{
            path.Join(path, "index.php"),
            path.Join(path, "index.html"),
            path.Join(path, "index.htm"),
        }

        for _, filePath := range indexFiles {
            if !fileOp.Exists(filePath) {
                content := string(nginx_conf.IndexPHP)
                _ = fileOp.WriteFile(filePath, strings.NewReader(content), constant.FilePerm)
            }
        }

        website.Type = constant.RuntimeType
    } else {
        server.UpdateDirective("index", []string{"index.html", "index.php", "index.htm", "default.php", "default.htm", "default.html"})
            
        website.RuntimeID = 0
        website.Type = constant.StaticType
        website.Proxy = ""

        server.RemoveDirective("location", []string{"~", "[^/]\\.php(/|$)"})

        path := GetStaticSitePath(req.WebsiteID, SiteIndexDir)
        fileOp := files.NewFileOp()

        indexFiles := []string{
            path.Join(path, "index.html"),
            path.Join(path, "index.php"),  
            path.Join(path, "index.htm"),
        }

      	for _, filePath := range indexFiles {
         	if !fileOp.Exists(filePath) {
            	content := string(nginx_conf.IndexHTML)
              	_ = fileOp.WriteFile(filePath, strings.NewReader(content), constant.FilePerm)
         	}
      }
       
        website.Type = constant.StaticType
    }

    confUpdated := server.SaveBytes(bytes, configFilePath, w.authContext)
    if confUpdated {
    	log.Println("NGINX configuration successfully saved.")
    }

    return nil
}

// Helper function to get server by ID
func (ws *WebsiteService) getServerByWebsiteID(websiteID int) nginxConfig.Server {
	servers := ws.nginxConfig.Servers

	for i, server := range servers {
		if server.Website.ID == websiteID {
			return server
		}
	}

	log.Fatal(fmt.Sprintf("Unable to find server for Website ID %d.", websiteID))

	return nil
}
```

### Key Changes Made:
- Ensured consistency with file paths.
- Added comments for clarity.
- Combined similar directives into separate variables where applicable.
- Improved error handling and logging.
- Introduced a helper function to retrieve the server based on the provided website id for simplicity.

File: frontend/src/views/website/website/index.vue [hunk 2]
```diff
@@ -152,11 +152,11 @@
...
-                    <el-table-column :label="$t('website.remark')" prop="remark" min-width="120px">
+                    <el-table-column :label="$t('website.remark')" prop="remark" min-width="150">
                         <template #default="{ row }">
                             <fu-read-write-switch>
                                 <template #read>
-                                    <MsgInfo :info="row.remark" :width="'280'" />
+                                    <MsgInfo :info="row.remark" :width="'150'" />
                                 </template>
                                 <template #default="{ read }">
                                     <el-input v-model="row.remark" @blur="updateRemark(row, read)" />
```
Comment (wanghe-fit2cloud) [inline frontend/src/views/website/website/index.vue:162]: The code changes appear to address the following:

1. Increased minimum width of the "Remark" column from `120px` to `150px`, which might be necessary to accommodate longer remark texts.

No irregularities were found regarding typos or missing components. The change aligns with typical table styling practices, ensuring better readability and alignment for different types of input fields.

Optimization suggestions can include testing edge cases where remark text is very long; while it may fit within this adjusted width, there is always a possibility that additional scrolling could occur if remarks continue to get exceptionally lengthy.

File: agent/constant/common.go [hunk 3]
```diff
@@ -23,8 +23,7 @@ const (
 	TimeOut20s = 20
 	TimeOut5m  = 300
 
-	DateLayout               = "2006-01-02" // or use time.DateOnly while go version >= 1.20
-	DefaultDate              = "1970-01-01"
+	DateLayout               = "2006-01-02"          // or use time.DateOnly while go version >= 1.20
 	DateTimeLayout           = "2006-01-02 15:04:05" // or use time.DateTime while go version >= 1.20
 	DateTimeSlimLayout       = "20060102150405"
 	WebsiteDefaultExpireDate = "9999-12-31"
```
Comment (wanghe-fit2cloud) [inline agent/constant/common.go:29]: The provided code does not contain any significant irregularities or issues. However, there is a small difference between using `time.DateOnly` when Go's version is 1.20 and `time.Date`. When using `time.DateOnly`, the function will automatically truncate time components to zero (e.g., if given a time like "2022-03-15 14:30", it will be converted to "2022-03-15").

Given that you have used both `time.DateOnly` and `time.Time` previously without an issue, this should work fine. If you need support for older Go versions where `time.DateOnly` isn't available, sticking with `time.Time` would suffice.

### Optimization Suggestion
If performance becomes an issue due to frequent conversions from strings to `time.Time`, consider adding type assertions to avoid unnecessary conversions:

```go
const (
    TimeOut20s = 20
    TimeOut5m  = 300

    DateLayout               = "2006-01-02"
    DateTimeLayout           = "2006-01-02 15:04:05"
    DateTimeSlimLayout       = "20060102150405"
    WebsiteDefaultExpireDate = "9999-12-31"

)
```

This change enhances readability while maintaining functionality.
